activation/activation.go

```diff
@@ -160,23 +160,44 @@ func NewBuilder(
 }
 
 func (b *Builder) proof(ctx context.Context, challenge []byte) (*types.Post, *types.PostInfo, error) {
+	started := false
+	retries := 0
 	for {
 		client, err := b.postService.Client(b.signer.NodeID())
-		if err == nil {
+		if err != nil {
+			select {
+			case <-ctx.Done():
+				events.EmitPostFailure()
+				return nil, nil, ctx.Err()
+			case <-time.After(2 * time.Second): // Wait a few seconds and try connecting again
+				retries++ // TODO(mafa): emit event warning user about lost connection after a few retries
+				continue
+			}
+		}
+		if !started {
 			events.EmitPostStart(challenge)
-			post, postInfo, err := client.Proof(ctx, challenge)
-			if err != nil {
+			started = true
+		}
```

Comment:
How about
```go
once.Do(func() {events.EmitPostStart(challenge)})
```

Reply:
Doesn't quite work with also only emitting `PostFailure` only after it started generating a post.